cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -76,6 +78,8 @@ func ValidateMasterConfiguration(c *kubeadm.MasterConfiguration) field.ErrorList
 	allErrs = append(allErrs, ValidateAbsolutePath(c.CertificatesDir, field.NewPath("certificates-dir"))...)
 	allErrs = append(allErrs, ValidateNodeName(c.NodeName, field.NewPath("node-name"))...)
 	allErrs = append(allErrs, ValidateToken(c.Token, field.NewPath("token"))...)
+	allErrs = append(allErrs, ValidateTokenUsages(c.TokenUsages, field.NewPath("token-usages"))...)
+	allErrs = append(allErrs, ValidateTokenGroups(c.TokenUsages, c.TokenGroups, field.NewPath("token-usages"))...)
```

Comment:
s/`token-usages`/`token-groups` ?

Actually, `token-usages`, `token-groups` and other field path names such as `cert-altnames` aren't defined anywhere else. I think we should use the json names defined in `cmd/kubeadm/app/apis/kubeadm/v1alpha1/types.go` instead. Otherwise users might get confused when they get a validation error. @kubernetes/sig-cluster-lifecycle-api-reviews  WDYT?

Reply:
I totally agree
I already fixed for what is in scope of this PR; I will send another PR for aligning alle the other field names to json names.